src/librustc/lint/builtin.rs

```diff
@@ -102,13 +102,13 @@ declare_lint! {
 
 declare_lint! {
     pub TRIVIAL_CASTS,
-    Warn,
+    Allow,
```

Comment:
We decided we would make the trivial cast lint Allow too until type ascription lands.